doc/pveclibmaindox.h

```diff
@@ -1104,59 +1215,1050 @@ s0 = vec_addeq (&c0, a0, b0, c1);
+endif
+	$(PVECCOMPILE) $(PVECLIB_POWER9_CFLAGS) -c -o $@ $(srcdir)/vec_runtime_PWR9.c
+endif
+* \endcode
+* We change the target (vec_staticrt_PWR9.lo) of the rule to indicate
+* that this object is intended for a <I>static</I> runtime archive.
+* And we list prerequisites vec_runtime_PWR9.c and
+* \$(pveclibinclude_HEADERS)
+*
+* For the recipe we expand both clauses (am__fastdepCC and AMDEP) from
+* the example. We don't know exactly what they represent or do, but
+* assume they both are needed for some configurations.
+* We use the alternative PVECCOMPILE to provide all the
+* libtool commands and options we need without the CFLAGS.
+* We use new PVECLIB_POWER9_CFLAGS macro to provide all the platform
+* specific target options we need. The automatic variable \$\@ provides
+* the file name of the target object (vec_staticrt_PWR9.lo).
+* And we specify the \$(srcdir) qualified source file
+* (vec_runtime_PWR9.c) as input to the compile.
+* We can provide similar rules for the other processor targets
+* (PWR8/PWR7).
+*
+* With this technique we control the compilation of specific targets
+* without requiring unique LTLIBRARIES. This was only required before
+* so libtool would allow target specific CFLAGS. So we can eliminate
+* libpvecPWR9.la, libpvecPWR8.la, and libpvecPWR7.la from
+* lib_LTLIBRARIES.
+*
+* Continuing the theme of separating the static archive elements from
+* DSO elements we rename libpveccommon.la to libpvecstatic.la.
+* We can add the common (none target specific) source files and CFLAGS
+* to <I>libpvecstatic_la</I>.
+* \code
+libpvecstatic_la_SOURCES = tipowof10.c decpowof2.c
+
+libpvecstatic_la_CFLAGS = $(AM_CPPFLAGS) $(PVECLIB_DEFAULT_CFLAGS) $(AM_CFLAGS)
+* \endcode
+* We still need to add the target specific objects generated by the
+* rules above to the libpvecstatic.a archive.
+* \code
+# libpvecstatic_la already includes tipowof10.c decpowof2.c.
+# Now add the name qualified -mcpu= target runtimes.
+libpvecstatic_la_LIBADD = vec_staticrt_PWR9.lo
+libpvecstatic_la_LIBADD += vec_staticrt_PWR8.lo
+libpvecstatic_la_LIBADD += vec_staticrt_PWR7.lo
+* \endcode
+* \note the libpvecstatic archive will contain 2 or 3 implementations
+* of each target specific function
+* (i.e. the function vec_mul128x128() will have implementations
+* vec_mul128x128_PWR7() and vec_mul128x128_PWR8(),
+* vec_mul128x128_PWR9()).
+* This OK because because the target suffix insures the name is unique
+* within the archive.
+* When an application calls function with the appropriate target suffix
+* (using the __VEC_PWR_IMP() wrapper macro) and links to libpvecstatic,
+* the linker will extract only the matching implementations and include
+* them in the static program image.
+*
+*
+* \subsection main_libary_issues_0_0_2 Building dynamic runtime libraries
+*
+* Building objects for dynamic runtime libraries is a bit more
+* complicated than building static archives. For one dynamic libraries
```

Comment:
wait. more complicated than the libtool mess above?

Reply:
Have you meet __attribute__ ((ifunc ("resolve_vec_mul512x512"))) yet?